File: Hotel_Neruda/ClientApp/src/app/Hotel_Neruda/models/reserva.ts
import { Habitacion } from "./habitacion";
import { Persona } from "./persona";

export class Reserva {
    id = 0;
    inicio = new Date();
    final = new Date();
    cliente = "";
    _cliente = new Persona();
    habitacion = 0;
    _habitacion = new Habitacion();
    checkIn = false;
    checkOut = false;
    base = 0;
    total = 0;
}

File: Entidad/Reserva.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Entidad
{
    public class Reserva
    {
        [Key]
        public int Id { get; set; }
        public DateTime Inicio { get; set; }
        public DateTime Final { get; set; }
        public string Cliente { get; set; }
        public int Habitacion { get; set; }
        public bool CheckIn { get; set; }
        public bool CheckOut { get; set; }
        public double Base { get; set; }
        public double Total { get; set; }
        public int Camarada { get; set; }
    }
}